torch/_dynamo/variables/tensor.py

```diff
@@ -402,6 +406,8 @@ def call_method(
                 ),
                 **options,
             )
+            tx.metadata_mutated_variables[self] = out
```

Comment:
This won't work with the checkpoint/restore logic.

You should do something similar to:
https://github.com/pytorch/pytorch/blob/428540001d9fff9c68ad2b24b8e7794800be6c77/torch/_dynamo/symbolic_convert.py#L497-L514

Though you might need to add a case where you compare `TensorVariable.proxy` instead of `.mutable_local`

Reply:
I added a function named update_locals_and_stack_ex for replacements.